File: app/src/test/java/com/github/goldy1992/mp3player/client/activities/EmptyMediaActivityCompatFragmentActivity.java
package com.github.goldy1992.mp3player.client.activities;

import android.os.Bundle;

import com.github.goldy1992.mp3player.dagger.components.DaggerTestMediaActivityCompatComponent;
import com.github.goldy1992.mp3player.dagger.components.TestMediaActivityCompatComponent;

public class EmptyMediaActivityCompatFragmentActivity extends MediaActivityCompat {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        initialiseDependencies();
        super.onCreate(savedInstanceState);
    }

    @Override
    String getWorkerId() {
        return "WORKER_ID";
    }

    @Override
    void initialiseDependencies() {
        TestMediaActivityCompatComponent component =
            (TestMediaActivityCompatComponent)
            DaggerTestMediaActivityCompatComponent
                .factory()
                .create(getApplicationContext(), getWorkerId(), this);
        component.inject(this);
        mediaActivityCompatComponent = component;
    }

    @Override
    boolean initialiseView(int layoutId) {
        return true;
    }
}

File: client/testsupport/src/main/java/com/github/goldy1992/mp3player/client/testsupport/activities/EmptyMediaActivityCompatFragmentActivity.kt
package com.github.goldy1992.mp3player.client.testsupport.activities

import android.os.Bundle
import com.github.goldy1992.mp3player.client.activities.MediaActivityCompat
import com.github.goldy1992.mp3player.client.testsupport.dagger.components.DaggerTestMediaActivityCompatComponent
import com.github.goldy1992.mp3player.client.testsupport.dagger.components.TestMediaActivityCompatComponent
import com.github.goldy1992.mp3player.commons.ComponentClassMapper

class EmptyMediaActivityCompatFragmentActivity : MediaActivityCompat() {

    override fun onCreate(savedInstanceState: Bundle?) {
        initialiseDependencies()
        super.onCreate(savedInstanceState)
    }

    override val workerId: String
        get() = "WORKER_ID"

    override fun initialiseDependencies() {
        val component = DaggerTestMediaActivityCompatComponent
                .factory()
                .create(applicationContext, workerId, this, ComponentClassMapper.Builder().build()) as TestMediaActivityCompatComponent
        component.inject(this)
        mediaActivityCompatComponent = component
    }

    override fun initialiseView(layoutId: Int): Boolean {
        return true
    }
}